src/NServiceBus.Core/Pipeline/PipelineSettings.cs

```diff
@@ -176,9 +185,28 @@ public void Register<TRegisterStep>() where TRegisterStep : RegisterStep, new()
         public void Register(RegisterStep registration)
         {
             Guard.AgainstNull(nameof(registration), registration);
+            EnsureWriteEnabled(nameof(registration), "register");
+
             modifications.Additions.Add(registration);
         }
 
+        /// <summary>
+        /// Locks the pipeline settings to prevent further modifications.
+        /// </summary>
+        internal void PreventChanges()
+        {
+            locked = true;
+        }
+
+        void EnsureWriteEnabled(string key, string operation)
```

Comment:
Should it be called 'Guard' as per existing guard methods? Even better, should it be an Guard-like method, e.g.

```
Guard.AgainstLockedSetting(string key, string operation)
```

Reply:
Could be, but I followed how SettingsHolder is implemented: https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Settings/SettingsHolder.cs#L308-L322